### PR TITLE
Updated performance tests to work with multidimensional support

### LINKED ISSUE #100: Update performance tests and remove duplication

### PR DESCRIPTION
**Pull request by melissawm**
This updates the performance tests to work with the new multidimensional dataset support. Test results are still pending.

Closes #100 and #87

### CODE REVIEW

**Comment by asmeurer** [comment]
Once https://github.com/deshaw/versioned-hdf5/pull/114 is merged you can merge master in and push the branch up to this repo and it will push the docs build to https://deshaw.github.io/versioned-hdf5/_docs-analysis.

File: analysis/performance_tests.py [hunk 1]
```diff
@@ -198,26 +198,28 @@ def save(self, summary, filename):
...
+    # tests = [test_small_fraction_changes_sparse,
+    #          test_large_fraction_constant_sparse]
+    # for test in tests:
+    #    testcase = test(num_transactions=[50, 100, 500, 1000, 5000, 10000],
+    #                    exponents=[14],
+    #                    compression=[None,])
+    #    summary, msg = testcase.create_files(versions=True)
+    #    testcase.save(summary, f"{testcase.testname}")
+    #    summary, msg = testcase.create_files(versions=False)
+    #    testcase.save(summary, f"{testcase.testname}_no_versions")
```
Comment (scopatz) [inline analysis/performance_tests.py:210]: Can this code justy be removed?

File: docs/performance.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+In order to analyze this, we will consider test files created with a variable number of versions (or transactions) of a dataset consisting of three ndarrays of variable length. One test includes a two-dimensional ndarray as part of this dataset, but all other test cases consist of three one-dimensional ndarrays per dataset.
+
+With each new version a certain number of rows is added, removed, and modified. For
+these tests, all changes are made to elements chosen according to a `power law
+<https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.powerlaw.html>`__
+which biases the modifications towards the end of the array, simulating a possible use case of modifying more recent results in a given timeseries.
+
+The tests are as follows:
+
+1. A large fraction of changes is made to the dataset with each new version: The dataset initially has three arrays with 5000 rows, and 1000 positions are chosen at random and changed, and a small number (at most 10) rows are added or deleted with each new version. We will refer to this test as `test_large_fraction_changes_sparse`.
```
Comment (scopatz) [inline docs/performance.rst:27]: Should these all use double backticks?

**Comment by scopatz** [comment]
Amazing writing here @melissawm! Thanks so much!

File: docs/performance.rst (hunk 2 above)
Comment (asmeurer) [reply docs/performance.rst:27]: We have single backticks configured to render as code. Though, in the future, we may want to write long form documents like this in Markdown.

File: analysis/performance_tests.py (hunk 1 above)
Comment (melissawm) [reply analysis/performance_tests.py:210]: I'll do it in another PR, sorry about that.

**Pull request merged**